msal/src/main/java/com/microsoft/identity/client/PublicClientApplicationConfigurationFactory.java

```diff
@@ -93,12 +95,14 @@ private static PublicClientApplicationConfiguration initializeConfigurationInter
             config.validateConfiguration();
         }
 
+        final PublicClientApplicationConfiguration configWithGlobal = globalSettings.mergeConfigurationWithGlobal(config);
+
         //Initialize internal library configuration
-        final LibraryConfiguration libraryConfiguration = LibraryConfiguration.builder().authorizationInCurrentTask((config.authorizationInCurrentTask())).build();
+        final LibraryConfiguration libraryConfiguration = LibraryConfiguration.builder().authorizationInCurrentTask((configWithGlobal.authorizationInCurrentTask())).build();
```

Comment:
I believe the motivation behind [that work item](https://identitydivision.visualstudio.com/Engineering/_workitems/edit/1400670) was that GlobalSettings would replace the `LibraryConfiguration`.